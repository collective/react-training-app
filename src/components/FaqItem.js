import React, { Component } from "react";
import PropTypes from "prop-types";
import "./FaqItem.css";

class FaqItem extends Component {
  static propTypes = {
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired
  };

  render() {
    return (
      <li className="faq-item">
        <h2 className="question">{this.props.question}</h2>
        <p>{this.props.answer}</p>
      </li>
    );
  }
}

export default FaqItem;
