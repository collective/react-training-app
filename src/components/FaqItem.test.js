import React from "react";
import renderer from "react-test-renderer";

import FaqItem from "./FaqItem";

describe("FaqItem", () => {
  it("renders a FAQ item", () => {
    const component = renderer.create(
      <FaqItem
        question="What is the answer to life the universe and everything?"
        answer="42"
      />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
