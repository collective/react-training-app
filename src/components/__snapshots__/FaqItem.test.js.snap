// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqItem renders a FAQ item 1`] = `
<li
  className="faq-item"
>
  <h2
    className="question"
  >
    What is the answer to life the universe and everything?
  </h2>
  <p>
    42
  </p>
</li>
`;
